examples/lines_bars_and_markers/linestyles.py

```diff
@@ -3,31 +3,44 @@
 Linestyles
 ==========
 
-This examples showcases different linestyles copying those of Tikz/PGF.
+This example showcases different linestyles copying those of Tikz/PGF.
+Linestyle can be provided as simple as *solid* , *dotted* or *dashed*.
+Moreover, the dashing of the line can be controlled by a dash tuple
+such as (offset, (on_off_seq)) as mentioned in `.Line2D.set_linestyle`.
+For e.g., ``(0, (3, 10, 1, 15))`` means 3pt-line,10pt-space,1pt-line,15pt-space
+with no offset.
+
+*Note*: The dash style can also be configured via `.Line2D.set_dashes`
+as shown in :doc:`/gallery/lines_bars_and_markers/line_demo_dash_control`
+and passing a list of dash sequences using the keyword *dashes* to the
+cycler in :doc:`property_cycle </tutorials/intermediate/color_cycle>`.
 """
 import numpy as np
 import matplotlib.pyplot as plt
 from collections import OrderedDict
 from matplotlib.transforms import blended_transform_factory
 
-linestyles = OrderedDict(
-    [('solid',               (0, ())),
+linestyles = [
+     ('solid',               ('-')),     # Same as (0, ())
+     ('dotted',              (':')),     # Same as (0, (1, 1))
+     ('dashed',              ('--')),
+     ('dashdot',             ('-.')),
+
      ('loosely dotted',      (0, (1, 10))),
-     ('dotted',              (0, (1, 5))),
      ('densely dotted',      (0, (1, 1))),
 
      ('loosely dashed',      (0, (5, 10))),
-     ('dashed',              (0, (5, 5))),
      ('densely dashed',      (0, (5, 1))),
 
      ('loosely dashdotted',  (0, (3, 10, 1, 10))),
-     ('dashdotted',          (0, (3, 5, 1, 5))),
      ('densely dashdotted',  (0, (3, 1, 1, 1))),
 
+     ('dashdotdotted',       (0, (3, 5, 1, 5, 1, 5))),
      ('loosely dashdotdotted', (0, (3, 10, 1, 10, 1, 10))),
-     ('dashdotdotted',         (0, (3, 5, 1, 5, 1, 5))),
-     ('densely dashdotdotted', (0, (3, 1, 1, 1, 1, 1)))])
+     ('densely dashdotdotted', (0, (3, 1, 1, 1, 1, 1)))]
 
+# Reverse the list so that plots and linestyles are in same order.
+linestyles = OrderedDict(linestyles[::-1])
 
 plt.figure(figsize=(10, 6))
```

Comment:
Not necessarily part of this PR, but since you're touching this code, would you be ok with putting it in OO format:
```python
fig, ax = plt.subplots()
```

Reply:
It is a good idea. I will update it with OO format.